### PR TITLE
Add server auto-discovery

### PR DESCRIPTION
**Pull request by oddstr13**
image

Discovery tested to be working on emulators v1.2.0, v2.0.0, v3.0.0, v4.0.0 & v5.0.0, as well as my v3.something TV

### CODE REVIEW

File: org.jellyfin.webos/js/index.js [hunk 1]
```diff
@@ -467,3 +467,102 @@ window.addEventListener('message', function (msg) {
             break;
     }
 });
+
+
+
+
```
Comment (Artiume) [inline org.jellyfin.webos/js/index.js:473]: lots of blank columns, this intentional? Also notice we don't have any lint enforcement running yet :(

Comment (oddstr13) [reply org.jellyfin.webos/js/index.js:473]: I didn't do a linting pass (this is just a visual break, should probably have added a comment here).
I don't know the tools to set up linting for js, css and html.

**Comment by EraYaN** [comment]
I'll go and try this on my TV tonight, and then 🚀 !

**Comment by EraYaN** [comment]
This seems to work (v4.5 something TV), but it can only discover the IP address, I'd be good if it saved (and connected) using the hostname as well if we expose that anywhere. Since otherwise it only really works in static IP scenarios. I now have another issue while playing that it "crashes" to the login page during playback which is weird, but should be unrelated.

This PR should increase the version number though. (0.3.0 seems in order with this change)

File: org.jellyfin.webos/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "org.jellyfin.webos.service",
+  "version": "1.0.0",
```
Comment (EraYaN) [inline org.jellyfin.webos/package.json:3]: And this should probably match the version of the app in general

**Comment by oddstr13** [comment]
The address is whatever the server responds with, so if you want domain name, that needs to be implemented here;

https://github.com/jellyfin/jellyfin/blob/1894239fee0258af70a4cbaa1f79fdc4a6cb4030/Emby.Server.Implementations/Udp/UdpServer.cs#L50-L52

I have not seen it returning to the login page unless going via the select server option in the menu. Or, if you mean the username/password entry page, then that would be jellyfin-web, not the code in this repo.

Version set to 0.3.0 in both files.

**Comment by Artiume** [comment]
so would we want to make that configurable like urloverride https://jellyfin.org/docs/general/administration/configuration.html#main-configuration-options or would it be automatic? i am used to having provide the domain to which the server is discoverable at since the service itself doesnt typically know its domain

**Review by Artiume — APPROVED** [review]

**Comment by EraYaN** [comment]
> I have not seen it returning to the login page unless going via the select server option in the menu. Or, if you mean the username/password entry page, then that would be jellyfin-web, not the code in this repo.

I mean the main screen the code in this repo creates. But that is probably a web error, just one that might block us from having it in the store.

**Review by EraYaN — APPROVED** [review]

**Pull request merged**